### PR TITLE
Upgrade oneDNN to v2.6.0

### PR DESCRIPTION
**Pull request by yanbing-j**
This PR upgrades oneDNN to v2.6.0.

v2.6.0 changes:

- Improved performance for future Intel Xeon® Scalable processors (code name Sapphire Rapids). The functionality requires Linux kernel 5.16 or later.
- Improved performance of matmul primitive for processors with Intel AVX-512 support.
- Introduced bfloat16 destination support for int8 convolution, matmul and inner product primitives for processors with Intel AVX-512 support and or future Intel Xeon® Scalable processors (code name Sapphire Rapids)
- Extended RNN primitive with support for [AUGRU cell](https://oneapi-src.github.io/oneDNN/dev_guide_rnn.html#augru).
- Added support for non-zero negative slope in ReLU post-op for batch normalization primitive.
- Introduced support for mixed source and destination data types in softmax primitive.
- Introduced [persistent cache API](https://oneapi-src.github.io/oneDNN/dev_guide_persistent_cache.html). This functionality allows to serialize and reuse JIT kernels.

### CODE REVIEW

**Review by frank-wei — APPROVED** [review]
LGTM!

**Comment by sanchitintel** [comment]
Hi @frank-wei, this PR uses a more recent version of oneDNN Graph, which is incompatible with the current master branch (#74596 was merged a couple of hours ago, which builds an older version of oneDNN Graph).

Sorry for the inconvenience, but this PR would've to be imported again, as [I added small changes in this PR](https://github.com/pytorch/pytorch/pull/75398/commits/9c7ef5507cc39d80f84b1746078d0d222500abf6) that enable support for the new oneDNN Graph version that's being added via this PR. These changes are in the PyTorch-oneDNN-Graph-integration code & are necessary because unlike the current version of oneDNN Graph that's being used in the master branch, the oneDNN Graph version used in this PR wouldn't use blocked layout (and only use strided-layout instead) for tensors that are output of one oneDNN Graph partition, and an input to another. It'd be great if @eelison could take a look at these changes, whenever possible.

Thank you!

**Comment by frank-wei** [comment]
Hi, @sanchitintel ,
I am ok with it since it is a hot fix. Ping @eellison for review comment.
I will import first to check the internal pass.

**Comment by frank-wei** [comment]
@sanchitintel  could you solve the branch conflicts?

**Comment by sanchitintel** [comment]
Hi @frank-wei, unfortunately, #74596 got reverted because a macOS test is failing. 😞 
I had misconstrued the macOS failure as a CI failure because https://github.com/pytorch/pytorch/runs/6204422214?check_suite_focus=true shows some AWS error.

Can you please wait to import & merge this PR until #74596 is re-landed? 
I'll ping you once it's done. Thank you!

**Comment by sanchitintel** [comment]
Hi @frank-wei, sorry for the inconvenience!
I've reverted the changes I made to this PR, so that it can be landed.
I'll add those changes to the oneDNN-Graph-integration instead.

**Review by frank-wei — APPROVED** [review]

**Review by bigfootjon — APPROVED** [review]
